src/controller/connexionController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+namespace TourCMS\OnBoarding\Controller;
+
+use Error;
+use TourCMS\OnBoarding\Model\connexion;
+
+class connexionController
+{
+    protected $connexion;
+
+    public function __construct($data = null)
+    {
+        $this->connexion = new connexion($data);
+    }
+
+    function redisDataInsertion($dataType, $data = null)
+    {
+        if (isset($data)) {
+            $this->connexion->dataSetter($data);
+        }
+
+        try {
+            foreach ($this->connexion->data as $key => $value) {
+                switch ($dataType) {
+                    case 'json':
+                        $this->connexion->redis->storeItemInRedis($key, json_encode($value), 'string');
+                        break;
+                    case 'array':
+                        $this->connexion->redis->storeItemInRedis($key, $value, 'array');
+                        break;
+                    case 'string':
+                        $this->connexion->redis->storeItemInRedis($key, $value, 'string');
+                        break;
+                    case 'set':
+                        $this->connexion->redis->storeItemInRedis($key, $value, 'set');
+                        break;
+                }
+            }
+        } catch (Error $e) {
+            echo $e->getMessage();
+            header('Location:http://' . $_SERVER['SERVER_NAME'] . '/error');
```

Comment:
Redirect is out of the scope of what the function should do.
Ideally, should return a true/false